activesupport/lib/active_support/dependencies.rb

```diff
@@ -467,8 +467,7 @@ def load_file(path, const_paths = loadable_constants_for_path(path))
         result = Kernel.load path
       end
 
-      autoloaded_constants.concat newly_defined_paths unless load_once_path?(path)
-      autoloaded_constants.uniq!
+      autoloaded_constants.merge(newly_defined_paths) unless load_once_path?(path)
```

Comment:
Is not this going to be significantly slower? A micro benchmark says it is ~1.5x slower that using an array.

```ruby
gem 'benchmark-ips'
require 'benchmark/ips'
require 'set'

array = [1]
set = Set.new([1])

Benchmark.ips do |b|
  b.report('array') { array.concat([1]); array.uniq! }
  b.report('set') { set.merge([1]) }
  b.compare!
end
```

```
Warming up --------------------------------------
               array   108.343k i/100ms
                 set    73.877k i/100ms
Calculating -------------------------------------
               array      1.493M (±14.2%) i/s -      7.259M in   5.017411s
                 set    956.655k (± 3.1%) i/s -      4.802M in   5.024529s

Comparison:
               array:  1492742.7 i/s
                 set:   956654.7 i/s - 1.56x  slower
```

Reply:
I also believe this will be slower. The other options I see are:
1. Create some new type of array that calls uniq directly when you add a new item, as the api for adding dependencies seems to be circumvented or not used as expected. 
2. Find all the places where dependencies are being injected by what ever is watching for changed files and find a better way to do so.
3. Do #2 and do something to the api so this cannot happen again.

I personally feel that there is an underlying data type problem here. Arrays are notably faster, because sets are really a api built on top of Ruby hashes, we need some sort of unique array that is not as slow as a set as calling uniq in one of the methods leads to a confusing api dependency that is clearly not being subscribed to.

Reply:
I think Set is fine for now since we also use a Set in `loaded` and `history`